src/NServiceBus.NHibernate.Tests/SynchronizedStorage/NHibernateAmbientTransactionSynchronizedStorageSessionTests.cs

```diff
@@ -57,25 +60,26 @@ public async Task It_does_not_commit_if_callback_throws()
                 var transportTransaction = new TransportTransaction();
                 transportTransaction.Set(Transaction.Current);
 
-                var adapter = new NHibernateSynchronizedStorageAdapter(SessionFactory, null);
+                //The open method creates the NHibernateLazyNativeTransactionSynchronizedStorageSession
+                var syncSession = new NHibernateSynchronizedStorageSession(new SessionFactoryHolder(SessionFactory));
+                var success = await syncSession.TryOpen(transportTransaction, new ContextBag());
+                Assert.IsTrue(success);
```

Comment:
I'm not a big fan of mixing Assertions with Arrange. How about moving the assertion down?

Reply:
Same as above, this one is not really an assertion but a way to fail test in a meaningful way.